### DISCUSSION TITLE
Image elements do not have explicit width and height

### DISCUSSION THREAD

**Discussion by rajatdhoot123** [tailwindlabs/tailwindcss#4658] [Help]
![image](https://user-images.githubusercontent.com/18528826/122163113-4306d380-ce92-11eb-91d4-3f30b87bb34e.png)
I ran the lighthouse on https://tailwindcss.com/ it was an issue with image height and width
`Image elements do not have explicit width and height`

the same issue is on my project which I am working on I want to fix the issue but not able to do that tried googling but no luck every time I use inline height and width property on image tailwind preflight overtakes

```
img, video {
    max-width: 100%;
    height: auto;
}
```

Can you please help me to fix this issue?

**Comment by duncan-p** [comment]
Hey Rajat,  I came across this recently on a project as well and the solution is to add `width` and `height` attributes to your image elements.

E.g, on tailwindcss.com , the avatar image is a 40x40px jpg, so if width and height attributes were added like this...

```
<img src="/avatar.jpg" alt="Adam" class="w-10 h-10" width="40" height="40" loading="lazy" />
```

Re-running Lighthouse, the `Image elements do not have explicit width and height` warnings should be gone and the score should improve a little.

The image will still display at the size you specify in the tailwind width and height classes (i.e, `w-10`, `h-10` in this example) but it allows the browser to examine the width and height attributes before the image has loaded to know what aspect ratio to expect. This means it can *leave room* for the image, ready for when it's finished loading.  Doing this will help you get a better score in Lighthouse for CLS (cumulative layout shift).  

There's a good post about this on web.dev if you want to read more  
https://web.dev/optimize-cls/#images-without-dimensions

**Reply by rajatdhoot123** [reply]
Thanks, @duncan-p yes by implementing the suggestion it starts working will check the post as well.
